dojo/filters.py

```diff
@@ -354,6 +358,25 @@ def __init__(self, *args, **kwargs):
                 if exclude:
                     self.form.fields[field].label = "Not " + self.form.fields[field].label
 
+    def filter_queryset(self, queryset):
+        qs = super().filter_queryset(queryset)
+        try:
+            if hasattr(self, "form") and hasattr(self.form, "cleaned_data"):
+                for name, f in self.filters.items():
+                    field_name = getattr(f, "field_name", "") or ""
+                    # filtering on tag names would result duplicate rows, one for each matching tag
+                    if "tags__name" in field_name:
+                        value = self.form.cleaned_data.get(name, None)
+                        if value not in (None, "", [], (), {}):
+                            # distinct has a performance impact, so only apply it if needed.
+                            # we considered Postgress' DISTINCT ON, but it would enforce ordering by id
+                            # we considered changing to an EXISTS subquery, but it would make
+                            # our code dependant on the some of the django-tagulous internal
+                            return qs.distinct()
+        except Exception:
```

Comment:
What exceptions are expected here? Shouldn't we avoid these broad excepts? Especially as it's also returning a .distinct() (in case that would be the source for unexpected exceptions)